addons/hr_timesheet/tests/test_timesheet.py

```diff
@@ -407,6 +425,12 @@ def test_task_with_timesheet_project_change(self):
 
         self.assertEqual(timesheet.project_id, project, 'The project_id of timesheet shouldn\'t have changed')
 
+    def test_compute_display_name(self):
+        try:
+            self.timesheet.with_user(self.timesheet_manager_no_project_user)._compute_display_name()
+        except AccessError:
+            self.fail("AccessError raised despite use of sudo() in compute method.")
```

Comment:
I would prefer to test the normal flow and assert the display name is correctly computed instead of using try/except. If an AccessError exception is raised no need to cache it to say the test fails, the test will fail because of that exception raised.